### ISSUE TITLE
failed to load workspace due to scraping home directory for mod files

### ISSUE THREAD

**Issue by tafkam** [turbot/steampipe#2321]
**Describe the bug**
Steampipe seems to scrape the whole home directory for mods, and fails if it finds unexpected files, e.g. in git repositories:

```
% steampipe check all
Error: failed to load workspace: Failed to load all mod source files
Invalid character: The ";" character is not valid. Use newlines to separate arguments and blocks, and commas to separate items in collection values.
(/Users/MyUsername/git/db_scripts/sql/whatever-query-script.sp:4,57-58)

```

Changing the workspace (--workspace-chdir="/Users/MyUsername/steampipe") doesnt help with this.

**Steampipe version (`steampipe -v`)**
v0.15.4

**To reproduce**
Steps to reproduce the behavior (please include relevant code and/or commands).

**Expected behavior**
please don't search through everything in my home directory, and load whatever files are stored there

**Additional context**
error occurs with a fresh brew install on macOS

**Comment by kaidaguerre** [comment]
Hi @tafkam 

Thanks for raising this, sorry you are having issues. The mod file loading behaviour is as follows:

If a `mod.sp` file containing a mod definition is found in the working directory (the current directory or the `workspace-chdir` specified in the args) then:
- a recursive search is executed to find any `.sp` underneath the working directory. These  are parsed and loaded into the workspace
- any `.sql` files found recursively under the working directory are converted into `query` resources in the workspace

If NO `mod.sp` file is found:
- `.sp` files in the working directory ONLY are loaded into the workspace 


So it looks as though you must have a `mod.sp` in your home directory (or whatever directory you are running from)

> Changing the workspace (--workspace-chdir="/Users/MyUsername/steampipe") doesnt help with this.
This is surprising, as it seems to be working fine for me:

Running from a root folder containing a `mod.sp` - it recursively loads all sp and sql files, and the load fails 
```
Kais-MacBook-Pro:turbot kai$ sp query
Error: failed to load workspace: Failed to load mod
Unsupported block type: Blocks of type "report" are not expected here.
(/Users/kai/Dev/github/turbot/steampipe-mod-reports-poc-wip/reports/linking.sp:1,1-7)
```
Running from same folder with workspace-chdir pointing to a valid mod folder
```
Kais-MacBook-Pro:turbot kai$ sp query --workspace-chdir /Users/kai/Dev/github/turbot/steampipe/pkg/steampipeconfig/testdata/mods/single_mod_one_query
Welcome to Steampipe v0.16.0-rc.9
For more information, type .help
> 
```

What is the contents of `Users/MyUsername/steampipe`? Could you share the command you are trying and the error you receive.






What is the conty

**Comment by kaidaguerre** [comment]
Ah, a slight correction to the above 
```
If a mod.sp file containing a mod definition is found in the working directory (the current directory or the workspace-chdir specified in the args)
```

In fact, when looking for a modfile, it also searches _up_ the directory tree. So if you are running Steampipe from within a child folder of a mod folder, it recognises that mod as the current workspace mod (similar so how git operates - you can run git commands from child folders)

So in your case, if you have a mod file in your home directory, this would be loaded when running from any child folder. So the best suggestion is to remove the top level mod.sp file

Hope that make sense - please ping me if you need clarification

**Comment by tafkam** [comment]
Yes I did a mod init in my home directory, the mod functionality was a bit unclear for me, I just wanted to try out a dashboard which required a mod init. 
I've created a new steampipe directory, with a new mod.sp file in it, and use the env var to use that as working dir. 
This behaviour with the mods is a bit unexpected when you have no clue about the app and just want to try it out a bit. I also haven't found any hints to this behaviour in the documentation, which makes it a bit unsettling when steampipe closes with an error referencing to your git repositories.
